src/raftstore/store/peer.rs

```diff
@@ -539,11 +523,9 @@ impl Peer {
                     // the old leader may be expired earlier than usual, since a new leader
                     // may be elected and the old leader doesn't step down due to
                     // network partition from the new leader.
-                    // For lease safety during leader transfer, mark `leader_lease_expired_time`
-                    // to be unsafe until next_lease_expired_time from now
-                    self.leader_lease_expired_time = Some(Either::Right(
-                        self.next_lease_expired_time(monotonic_raw_now()),
-                    ));
+                    // For lease safety during leader transfer, transit `leader_lease`
+                    // to suspect.
+                    self.leader_lease.suspect(monotonic_raw_now());
```

Comment:
I think it should be `self.leader_lease.next_expired_time(monotonic_raw_now())`.